src/main/java/seedu/address/logic/commands/RequiredCommand.java

```diff
@@ -0,0 +1,194 @@
+    }
+
+    /**
+     * Retrieves the currentModules attribute of an Required Command object.
+     * @return currentModules attribute of type ObservableList<Module/>.
+     */
+    public ObservableList<Module> getCurrentModules() {
+        return currentModules;
+    }
+
+    /**
+     * Sets the argument modules as the attribute currentModules.
+     * @param modules target argument of type ObservableList<Module/>.
+     */
+    public void setCurrentModules(ObservableList<Module> modules) {
+        currentModules = modules;
+    }
+
+    /**
+     * Checks if a Module already exists in a given Module List.
+     * @param module Module that you wish to check.
+     * @param modules Module List that you wish to check against.
+     * @return True if the module already exists, false otherwise.
+     */
+    public boolean doesModuleAlreadyExist(Module module, ObservableList<Module> modules) {
+        for (Module mod : modules) {
+            if (module.isSameModule(mod)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Cross references the user's current list of Modules against the given
+     * modules argument and marks out any undone Modules. Displays a successMessage
+     * if all modules are done, and a failMessage if there are left over modules.
+     * @param modules List of Modules of a certain category (Eg. Foundation, IT Professsionalism)
+     * @param failMessage Fail message for particular category of Modules.
+     * @param successMessage Success message for particular category of Modules.
+     */
+    public void compareModules(ObservableList<Module> modules, String failMessage, String successMessage) {
+        boolean areModulesCleared = true;
+        StringBuilder modulesToAdd = new StringBuilder();
+        for (Module module : modules) {
+            if (!doesModuleAlreadyExist(module, currentModules)) {
+                String moduleToAdd = module.getModuleCode() + " (" + module.getModularCredits() + " MCs)";
+                modulesToAdd.append("\n").append(moduleToAdd);
+                areModulesCleared = false;
+            }
+        } if (areModulesCleared) {
+            leftOverModules += successMessage + "\n";
+        } else {
+            leftOverModules += failMessage + modulesToAdd + "\n";
+        }
+        leftOverModules += "\n";
+    }
+
+    /**
+     * Cross references the user's current list of Modules and marks out
+     * any undone Science Modules.
+     */
+    public void compareScience(ObservableList<Module> requiredScience) {
+        boolean areModulesCleared = true;
+        for (Module module : requiredScience) {
+            if (doesModuleAlreadyExist(module, currentModules)) {
+                leftOverModules += MESSAGE_SUCCESS_SCIENCE + "\n";
+                areModulesCleared = false;
+                break;
+            }
+        } if (areModulesCleared) {
+            leftOverModules += MESSAGE_SCIENCE + "\n";
```

Comment:
Likewise if `areModulesCleared` is true then shouldn't we be showing MESSAGE_SUCCESS_SCIENCE instead since the user has cleared this basket HAHA